### DISCUSSION TITLE
Komunikacja app service - azure vm

### DISCUSSION THREAD

**Discussion by galante13** [AzureCommunityPL/group#243] [Pomoc] [answered]
Mam wystawione api aplikacyjne na app service, które strzela do api wystawionego na azure vm (dokładnie chodzi o generowanie pdfa przy użyciu wkhtmltopdf). Chciałbym zablokować publiczny dostęp do tej vmki, a jedynie zezwolić na komunikację app service -> azure vm wewnątrz azure. Jak to zrobić?

**Comment by ptrstpp950** [comment, marked as answer]
- Po pierwsze od jakiegoś czasu `wkhtmltopdf` możesz mieć w Azure Web App - mało pomocne ale może juz o tym wiesz :)
- Drugie rozwiązanie to Azure Web App umożliwia połączenie po VNET (https://docs.microsoft.com/en-us/azure/app-service/web-sites-integrate-with-vnet) i możesz mieć tą VM'kę w VNET.
- Trzecie patrzysz na outbound ip i limitujesz dostęp po tych IP. Niby nie wszystko, ale większość ruchu wytniesz.

**Reply by galante13** [reply]
Ad. 1 Wiem, niestety fonty są nieobsługiwane, dlatego przeniosłem na vm.

Ad. 2 i 3
Właśnie to ogarniam, dzięki.

**Reply by mgrabarz** [reply]
@galante13 odrobina samopromocji z mojej strony - łap linka, pomoże: https://www.youtube.com/watch?v=OgotvrIBgnA

Oczywiście rozwiązaniem jest "AppService Vnet Integration", który wymaga wersji Standard (lub wyżej) AppServisu. Trzecie rozwiązanie Piotrka to takie nieco śliskie bym powiedział ;)

**Comment by rafal-linowiecki** [comment]
Proponuję zajrzeć tutaj: https://docs.microsoft.com/en-us/azure/virtual-network/virtual-networks-overview - dodaj api + vmkę do vneta i masz zamknęte z zewnątrz

PS. Nie lepiej postawić azure functions dla wkhtmltopdf  zamiast vmki?

**Reply by galante13** [reply]
Myślałem o tym, ale obawiam się, że będzie taki sam problem jak w przypadku appservice (fonty).
